### PR TITLE
build(aio): upgrade to @angular/material#2.0.0-beta.3

### PR DESCRIPTION
**Pull request by IgorMinar**

### CODE REVIEW

**Comment by IgorMinar** [comment]
for some reason I'm now getting 

```
core.es5.js:1085 ERROR Error: No provider for LiveAnnouncer!
    at injectionError (core.es5.js:1232) [angular]
    at noProviderError (core.es5.js:1270) [angular]
    at ReflectiveInjector_.webpackJsonp.3j3K.ReflectiveInjector_._throwOrNull (core.es5.js:2772) [angular]
    at ReflectiveInjector_.webpackJsonp.3j3K.ReflectiveInjector_._getByKeyDefault (core.es5.js:2811) [angular]
    at ReflectiveInjector_.webpackJsonp.3j3K.ReflectiveInjector_._getByKey (core.es5.js:2743) [angular]
    at ReflectiveInjector_.webpackJsonp.3j3K.ReflectiveInjector_.get (core.es5.js:2612) [angular]
    at AppModuleInjector.get [as _MdSnackBar_74] (app.module.ngfactory.ts:321) [angular]
    at AppModuleInjector.get [as _SwUpdateNotificationsService_77] (app.module.ngfactory.ts:333) [angular]
    at AppModuleInjector.webpackJsonp.kke6.AppModuleInjector.getInternal (app.module.ngfactory.ts:492) [angular]
    at AppModuleInjector.webpackJsonp.3j3K.NgModuleInjector.get (core.es5.js:3576) [angular]
    at resolveDep (core.es5.js:10981) [angular]
    at createClass (core.es5.js:10849) [angular]
    at createDirectiveInstance (core.es5.js:10675) [angular]
    at createViewNodes (core.es5.js:12024) [angular]
```

when I try to build this with `ng serve --prod --sm` (I manually disabled minification to see what can't be found, otherwise the error would say: `unknown provider for t`)

@gkalpak any idea why?

**Comment by gkalpak** [comment]
I have no idea why (especially since it works for JiT but not AoT 😕).

**Comment by gkalpak** [comment]
If I had to guess I would say it is somehow related to the fact that `MdSnackBar` dependes on `LiveAnnouncer` and `MdSnackBar` (with SkipSelf) and `LiveAnnouncer` requires `LiveAnnouncer` (with SkipSelf). Possibly in combination with fact that `LiveAnnouncer` is also aliased as `MdLiveAnnouncer`.

As a work-around, you can:

```ts
import { LiveAnnouncer, MdLiveAnnouncer } from '@angular/material';
...
providers: [
  { provide: LiveAnnouncer, useExisting: MdLiveAnnouncer },
  ...
```

**Comment by IgorMinar** [comment]
I'm suspecting that it's a regression in material or some kind of issue in the compiler.

**Review by gkalpak — CHANGES_REQUESTED** [review]
LGTM except for the `LiveAnnouncer` error :grin:

**Comment by IgorMinar** [comment]
@jelbourn thinks that this is going to be fixed with https://github.com/angular/material2/pull/3807

but I suspect that this is an Angular compiler or metadata collector bug, so I asked @tbosch and/or @chuckjaz to take a look.

**Comment by DevVersion** [comment]
@IgorMinar Also to add - I just noticed while investigating that the DI error only shows up inside of other providers. See https://github.com/angular/material2/pull/3807#issuecomment-290484367.

**Comment by petebacondarwin** [comment]
@IgorMinar I am assuming that this is still WIP - so added the appropriate label. Please update when it is ready again for review.

**Comment by gkalpak** [comment]
This raises the question again: Shouldn't we run e2e tests in `--prod` mode (at least on CI), so that we catch issues like this one (which only happen with AoT)?

**Comment by fxck** [comment]
@gkalpak that workaround breaks JIT though

**Comment by gkalpak** [comment]
I can't imagine why, but since https://github.com/angular/material2/pull/3807 has been merged, I guess there is not need for a work-around any more :smiley:

**Comment by IgorMinar** [comment]
I'm updating this PR now that  angular/material2#3807 was merged.

**Comment by IgorMinar** [comment]
ok. this PR should be ready to go now. @wardbell can you take a look plz?

**Review by gkalpak — APPROVED** [review]
I'm no @wardbell, but LGTM :smiley:

File: aio/src/styles/main.scss [hunk 1]
```diff
@@ -1,8 +1,8 @@
 // import global themes
-@import '~@angular/material/core/theming/all-theme';
+@import '~@angular/material/_theming.scss';
```
Comment (fxck) [inline aio/src/styles/main.scss]: no need for neither `_` and `.scss`

**Pull request merged**